Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/BaseListboxController.java

```diff
@@ -477,22 +478,41 @@ public void rename() throws InterruptedException {
     protected void share() {
         try {
             if (getSelectionCount() == 0) {
-                Messagebox.show("Please select a log or model to share", "Attention", Messagebox.OK, Messagebox.ERROR);
+                Notification.error("Please select a log or model to share");
                 return;
             } else if (getSelectionCount() > 1) {
-                Messagebox.show("You can not share multiple selections", "Attention", Messagebox.OK, Messagebox.ERROR);
+                Notification.error("You can not share multiple selections");
                 return;
             }
             Object selectedItem = getSelection().iterator().next();
             if (selectedItem instanceof FolderType) {
-                Messagebox.show("You can only share a log or model", "Attention", Messagebox.OK, Messagebox.ERROR);
+                Notification.error("You can only share a log or model");
                 return;
             }
-            Map arg = new HashMap<>();
-            arg.put("selectedItem", selectedItem);
-            arg.put("currentUser", UserSessionManager.getCurrentUser());
-            Window window = (Window) Executions.getCurrent().createComponents("macros/share.zul", null, arg);
-            window.doModal();
+            SummaryType summaryType = (SummaryType) selectedItem;
+            UserType userType = UserSessionManager.getCurrentUser();
+            String userName = userType.getUsername();
+            Group userAsGroup = getSecurityService().getGroupByName(userName);
+            Map<Group, AccessType> groupAccessMap;
+            Integer selectedItemId = summaryType.getId();
+            if (summaryType instanceof ProcessSummaryType) {
```

Comment:
Please dont use instanceOf.

https://armedia.com/blog/instanceof-avoid-in-code/

Reply:
@karthik-balasubramanian I think in this context, instanceof is simply used for type checking to verify allowable actions for the object of interest. Could you suggest an alternative to do that? We can introduce some kind of enum-based typing but instanceof seems to be the most efficient (https://stackoverflow.com/revisions/26514984/10).

Reply:
Using instanceOf operator is not object oriented programming. We should create an interface that has an abstract method, say getAllowableAction that returns the groupAccessMap. The summary type can implement that interface and each summary type will provide the appropriate access type.